remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java

```diff
@@ -481,6 +481,8 @@ private boolean shouldIntercept(Object key) {
 
    @Override
    public Object visitComputeCommand(InvocationContext ctx, ComputeCommand command) {
+      if (command.hasAnyFlag(FlagBitSets.ROLLING_UPGRADE))
+         return invokeNext(ctx, command);
```

Comment:
I think that theoretically we should run a reindex or index update on all possibly impacted indexed caches that use the entities that have changed. 
Since I think we have never covered this case so far, I think that it is safe to log something.
We should consider to open a separate issue to address the topic here, that if don't get wrong, it is about to support reindex on rolling upgrades.

Reply:
Hmm that's an interesting point. In the case of a rolling upgrade in the Operator, the "___protobuf_metadata" cache is always sync'd from the source -> target cluster before the user caches in our testsuite as they all have lower-case names. If I change the cachename to upper case, then the TreeSet will order the cache before the "___protobuf_metadata" cache and cache creation will fail as entities won't yet exist on the target cluster. I've created https://github.com/infinispan/infinispan-operator/issues/1988 to address this.

If the Operator ensures that the "___protobuf_metadata" data is always sync'd before the user caches are created, I assume that there's no need or any of the user caches to be reindexed?

Reply:
> Since I think we have never covered this case so far, I think that it is safe to log something.

My only issue with logging something is that in the Operator case, the user won't need to do anything due to the order of events described in my previous comment, so the warning logs will probably confuse the user as no action is actually required in that case.

Instead of logging a warning, how about we create a new issue to update the HR rolling upgrade docs so that we explicitly state that users with indexed caches must sync the "___protobuf_metadata" cache before other caches? Or in the case that they're migrating to an already partially populated cluster (I don't think this use-case is common) they have to ensure they re-index the cache after upgrade?

Reply:
creating a new issue makes totally sense to me. thanks

Reply:
I've created [ISPN-15696](https://issues.redhat.com/browse/ISPN-15696) for the documentation and added a commit to this PR.

Reply:
thanks!